### DISCUSSION TITLE
Chlorinator Relay, Pumps, and Circuits -  config issues.

### DISCUSSION THREAD

**Discussion by jamestalmage** [tagyoureit/nodejs-poolController#853] [REM]
So,
According to what I understand to be best practice... 

- Old school way of using single speed pumps / chlorinators was to hook them both to the same relay.
- Variable speed pumps should NOT be hooked to a relay.
- Chlorinators still should be.

This is what I've done. My IC40 is hooked to a pump relay, but my pump is powered 24/7, and controlled via RS485

The issue is that I'm not understanding how to tell njsPC that there's a relay for my chlorinator that isn't related to my pump...

I guess I really don't understand what "Circuits" are...
<img width="644" alt="Screenshot 2023-09-18 at 6 25 08 PM" src="https://github.com/tagyoureit/nodejs-poolController/assets/4082216/143a1386-bd50-4ebe-b28d-7a09fc1b549c">

<img width="645" alt="Screenshot 2023-09-18 at 6 27 00 PM" src="https://github.com/tagyoureit/nodejs-poolController/assets/4082216/4b3f7165-985d-47b1-a7fa-ab81248b9290">

**Comment by tagyoureit** [comment]
A circuit (your top picture) are physical relays.  Features are for turning on/off items that don't require a physical relay (like different speeds).  Pump circuits are assigned to the pump and tell the pump what speed to run at when a circuit/feature is active.

Your chlorinator should be hooked up to the Pool circuit.  Anytime the pool is on, the chlorinator will be powered.  

What is "incoming water"?

**Reply by jamestalmage** [reply]
Incoming water is the water temperature sensor. Probably a misconfiguration.

You're saying I should put the chlorinator relay there? And just delete the chlorinator circuit?

How would I then turn on the pump without turning on the chlorinator? Am I just relying on RS-485 to set it to zero when it isn't needed?

**Reply by tagyoureit** [reply]
> How would I then turn on the pump without turning on the chlorinator? Am I just relying on RS-485 to set it to zero when it isn't needed?

Yes, that's how it is supposed to work.  The relay is a secondary safety mechanism that cuts power when the pool is off, but otherwise the chlorinator can safely be powered and run anywhere from 0-100%.  Our [wiki](https://github.com/tagyoureit/nodejs-poolController/wiki/DIY-Standalone-Nixie-Pool-Controller#high-current-circuits) has a ton of great info in it.  The section I linked to talks about the filter (aka pool) circuit and wiring the chlor to that.

For temp sensors, you can see [these instructions](https://github.com/rstrouse/relayEquipmentManager/wiki/Temperature-Sensors).

**Reply by jamestalmage** [reply]
Got it.

Yes, I found the Wiki very thorough, at least as far as equipment selection and physical install. I've found it less helpful when it comes to the software configuration, not because the information isn't there, mostly because it feels spread out. (Like, I'm just realizing there's a whole other Wiki on the REM repository that has some explanations that would have been helpful, despite the fact that the big "REM/CHEM controller" wiki pages I used to build the thing are in the njsPC repository).

My unsolicited two cents, from a new-to-the-community users perspective: I'd consider consolidating WiKi and discussions to one repository. (Maybe even issues? Or at least encourage users who aren't sure where an issue belongs to open it on the main repository). Just a thought.

**Reply by tagyoureit** [reply]
I don't disagree.  It is helpful to separate out the physical hardware additions from the purely software based because of the types of installations and configurations that folks have.  But some of the core concepts are sprinkled throughout. 

Feel free to take a first pass.  😉

**Reply by CodeSlinger69** [reply]
The software side was the hardest for me to figure out as well. This might help: https://github.com/tagyoureit/nodejs-poolController/discussions/801

**Reply by spifftek70** [reply]
> A circuit (your top picture) are physical relays. Features are for turning on/off items that don't require a physical relay (like different speeds). Pump circuits are assigned to the pump and tell the pump what speed to run at when a circuit/feature is active.
> 
> Your chlorinator should be hooked up to the Pool circuit. Anytime the pool is on, the chlorinator will be powered.
> 
> What is "incoming water"?

What about shared bodies?  How do you apply settings for that?

**Comment by spifftek70** [comment]
So if I'm reading the discussion properly, the wireline diagram below is sufficient with the correct circuits settings in place?

![RS-485 drawio](https://github.com/user-attachments/assets/bfcf9d68-f1db-44bc-9349-37dcf70b8036)

**Reply by tagyoureit** [reply]
Chlor and pump are on the same circuit breaker, yes?

And the 24v is AC?  If so, looks good.

**Reply by spifftek70** [reply]
So ALL the relays connect to 24 V AC?  Or only the 3HP relay for a chlorinator?  I ask because the [Nixie diagrams](https://github.com/tagyoureit/nodejs-poolController/wiki/DIY-Standalone-Nixie-Pool-Controller#assemble-the-pool-controller-enclosure) show 24V DC connections.

Actually as it is shared body, I believe it would be more like this:
![redraw ic-40](https://github.com/user-attachments/assets/90cd22cd-1575-4993-8412-2b1e56175add)

**Reply by tagyoureit** [reply]
The relays are driven by 24vdc.  The equipment that is switched is AC.  And you want the intellichlor to be switched by the same relay that  provides power to the pump.  See attached from the EasyTouch installation manual.

![image](https://github.com/user-attachments/assets/52ee9199-e13a-4c12-84a0-e13c13706cad)
